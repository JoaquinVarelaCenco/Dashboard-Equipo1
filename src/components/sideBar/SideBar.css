/* code css */
.sideBar {
  min-width: 296px;
  position: fixed;
  height: 100vh;
  display: flex;
  flex-direction: column;
  /* align-items: center; */
  padding: 16px;
  justify-content: space-between;
  transition: all 0.5s ease-in-out;
  z-index: 99;
  animation: aparecer 0.5s linear;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.25);
}
@keyframes aparecer{
  0%{transform: translateX(-600px);}
  100%{transform: translateX(0px)}
}
.sideBar__img-logo {
  width: 80%;
  margin-bottom: 30px;
}
.sideBar__Links {
  display: flex;
  flex-direction: column;
  gap: 10px;
}
.sideBar__user-button {
  width: 100%;
  display: flex;
  border-radius: 16px;
  border: 0;
  padding: 3px;
  /* padding-left: 13px; */
  align-items: center;
  /* justify-content: center; */
}
.sideBar_eachLink {
  display: flex;
  align-items: center;
  text-decoration: none;
  gap: 25px;
  font-size: 15px;
  /* margin-left: 15px; */
  padding: 6px 0;
}

.sideBar_eachLink img{
  padding-left: 13px;
}
.sideBar__user-name {
padding-left: 5px;
font-size: 15px;
}

.sideBar__user-img {
  height: 22px;
  width: 22px;
}
.active {
  background-color: rgba(0, 0, 255, 0.136);
  border-radius: 6px;
}


@media (max-width: 1024px) {
  .sideBar {
    display: none;
    opacity: .955;
  }
 .hide{ 
    display: flex;
  }
 .unhide{
    display: none;
  }
}


/* Dark Mode */

.sideBar_Link .clear{
  color: var(--clear-Color);
}

.sideBar_Link .dark{
  color: var(--black-Color);
}

.dark .sideBar_eachLink img{
  filter: invert();
} 