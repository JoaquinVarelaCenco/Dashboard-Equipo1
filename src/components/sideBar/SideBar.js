import { NavLink, Link } from "react-router-dom";
import "./SideBar.css";
import logo from "../../assets/images/MiEcommerce.png";
import casita from "../../assets/images/home.svg";
import profilePic from "../../assets/images/ProfilePic.png";
import pakage from "../../assets/images/package-variant-closed.svg";
import store from "../../assets/images/store.svg";
import { useRef } from "react";
import { useContext } from "react";
import { ThemeContext } from "../../context/ThemeContext";
import SwitchToggle from "../SwitchToggle/SwitchToggle";
import { SideBarContext } from "../../context/SideBarContext";
import Button from "../Button/Button";

const Sidebar = () => {

  const {visibility, hideSideBar} = useContext(SideBarContext)
  const { theme } = useContext(ThemeContext); 

  return (
    <div className={`sideBar ${theme} ${visibility}`}>

      <div>
        <img className="sideBar__img-logo" src={logo} />
        <div className="sideBar__Links">
          <NavLink onClick={hideSideBar} to="/home" className={`sideBar_eachLink ${theme}`}> <img src= {casita} />Inicio</NavLink>
          <NavLink onClick={hideSideBar} to="/products" className={`sideBar_eachLink ${theme}`}> <img src= {pakage} />Productos</NavLink>
          <NavLink onClick={hideSideBar} to="/stores" className={`sideBar_eachLink ${theme}`}> <img src= {store} />Tiendas</NavLink>
        </div>
      </div>

      <div>
        <SwitchToggle />
        

       <Link to={"/profile"} className={`sideBar_eachLink ${theme}`} >
          <Button
            classN={"sideBar__user-button"}
          >
            <div className="sideBar__user-button__profile">
              <img className="sideBar__user-img" src={profilePic} />
            </div>
            <div>
              <p className="sideBar__user-name">Olivia</p>
            </div>
          </Button>

        </Link>
      </div>
    </div>
  );
};

export default Sidebar;
