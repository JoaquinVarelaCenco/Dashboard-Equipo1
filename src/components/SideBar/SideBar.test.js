import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SideBar from './SideBar';

describe("Test formulario de productos", ()=>{
   
 
        let component;
        beforeEach(()=>{
            component = render(<SideBar />,
            {wrapper: BrowserRouter })
        })
    
        test("Se debe renderiza correctamente", ()=>{
           
            const { container } = component
            expect(container).toMatchSnapshot()
                })
  
    test("Los links deben mostrarse y dirigir al lugar esperado", ()=>{
        const buttons = screen.getAllByRole('link');
        const btnHome = buttons[0];
        const btnProductos = buttons[1];
        const btnStores = buttons[2]
        const btnProfile = buttons[3]
        
        expect(btnHome.getAttribute('href')).toBe('/home');
        expect(btnProductos.getAttribute('href')).toBe('/products');
        expect(btnStores.getAttribute('href')).toBe('/stores');
        expect(btnProfile.getAttribute('href')).toBe('/profile');
    })   
})