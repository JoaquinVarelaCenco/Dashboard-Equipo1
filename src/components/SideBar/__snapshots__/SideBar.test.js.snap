// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formulario de productos Se debe renderiza correctamente 1`] = `
<div>
  <div
    class="sideBar undefined undefined"
  >
    <div>
      <img
        class="sideBar__img-logo"
        src="MiEcommerce.png"
      />
      <div
        class="sideBar__Links"
      >
        <a
          class="sideBar_eachLink undefined"
          href="/home"
        >
           
          <img
            src="home.svg"
          />
          Inicio
        </a>
        <a
          class="sideBar_eachLink undefined"
          href="/products"
        >
           
          <img
            src="package-variant-closed.svg"
          />
          Productos
        </a>
        <a
          class="sideBar_eachLink undefined"
          href="/stores"
        >
           
          <img
            src="store.svg"
          />
          Tiendas
        </a>
      </div>
    </div>
    <div>
      <label
        class="switch"
      >
        <input
          type="checkbox"
        />
        <span
          class="slider round"
        />
      </label>
      <a
        class="sideBar_eachLink undefined"
        href="/profile"
      >
        <button
          class="sideBar__user-button"
        >
          <div
            class="sideBar__user-button__profile"
          >
            <img
              class="sideBar__user-img"
              src="ProfilePic.png"
            />
          </div>
          <div>
            <p
              class="sideBar__user-name"
            >
              Olivia
            </p>
          </div>
        </button>
      </a>
    </div>
  </div>
</div>
`;
